### PR TITLE
Add Create and List Metric Profile tests

### PR DESCRIPTION
**Pull request by shreyabiradar07**
## Description

This PR implements the test scenarios for Metric Profile APIs

Fixes # ([1228](https://github.com/kruize/autotune/issues/1228))

### Type of change

- [ ] Bug fix
- [X] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

- [X] New Test  - `test_create_metric_profile`, `test_list_metric_profiles`
REST API tests can be run using `pytest -s test_create_metric_profile.py --cluster_type minikube` in `local_monitoring` test suite

- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: minikube

## Checklist :dart:

- [X] Followed coding guidelines
- [X] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [X] Tests added or updated

## Additional information
Image - `quay.io/shbirada/list_metric_profile:api`

### CODE REVIEW

File: tests/scripts/helpers/kruize.py [hunk 1]
```diff
@@ -364,3 +364,76 @@ def list_metadata(datasource=None, cluster_name=None, namespace=None, verbose=No
...
+    print("URL = ", url)
+
+    response = requests.post(url, json=metric_profile_json)
+    print("Response status code = ", response.status_code)
+    print(response.text)
+    return response
+
+# Description: This function deletes the metric profile
+# Input Parameters: metric profile input json
+def delete_metric_profile(input_json_file, invalid_header=False):
```
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/helpers/kruize.py:386]: Implementation of delete MetricProifle API is currently in progress and trying to run other tests is resulting in `409` error : `"Validation failed: Metric Profile already exists:"` as metric profile name is unique field. Will share the test results once resolved.

File: tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    input_json_data = json.load(open(input_json_file, 'r'))
+
+    form_kruize_url(cluster_type)
+
+    metric_profiles = []
+
+    input_metric_profile_name = input_json_data['metadata']['name']
+
+    # Create metric profile using the specified json
+    num_exps = 100
```
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py]: Updated variable name to `num_metric_profiles`

File: tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    ("metadata", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("name", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("slo", ERROR_500_STATUS_CODE, ERROR_STATUS)
+]
+
+
+@pytest.mark.sanity
+def test_create_metric_profile(cluster_type):
+    """
+    Test Description: This test validates the response status code of createMetricProfile API by passing a
```
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py:41]: done

File: tests/scripts/helpers/list_metric_profiles_without_parameters_schema.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+list_metric_profiles_without_parameters_schema = {
```
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/helpers/list_metric_profiles_without_parameters_schema.py:17]: Added copyrights in `list_metric_profiles_schema.py` and `list_metric_profiles_without_parameters_schema.py` files

File: tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        # Validate the json against the json schema
+        errorMsg = validate_list_metric_profiles_json(metric_profile_json, list_metric_profiles_schema)
+        assert errorMsg == ""
+
+        metric_profiles.append(copy.deepcopy(json_data))
+
+        response = delete_metric_profile(temp_json_file)
+        print("delete metric profile = ", response.status_code)
+
+    # Write the profiles to the output file
```
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py:185]: done

File: tests/scripts/helpers/kruize.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/helpers/kruize.py:386]: `/deleteMetricProfile` API is now implemented in #1246 

MetricProfile Test results: 
[#1231_metric_profile_tests_minikube.zip](https://github.com/user-attachments/files/16491863/1231_metric_profile_tests_minikube.zip)

[the remark this answers is not in the document]
Comment (chandrams) [reply tests/scripts/helpers/kruize.py:386]: @shreyabiradar07  There is a lot of logging in the test output, can you check this?

File: tests/setup.log [hunk 6]
```diff
@@ -131,4 +131,4 @@ Every request from UI or curl will go to API server, this will figure out kruize
       }
     ]
   }
-]
\ No newline at end of file
+]
```
[the remark this answers is not in the document]
Comment (chandrams) [reply tests/setup.log:134]: @shreyabiradar07 Can you check if this is required?

**Comment by chandrams** [comment]
@shreyabiradar07 - As discussed, please add an end-to-end test to test the entire workflow. You can raise a separate PR for this.

**Comment by shreyabiradar07** [comment]
@chandrams here are the updated test results 
Jenkins [kruize_functional_tests/102](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_functional_tests/102/)
[PR_1231_metric_profiles_tests_minikube.zip](https://github.com/user-attachments/files/16578101/PR_1231_metric_profiles_tests_minikube.zip)
[PR_1231_metric_profile_tests_openshift.zip](https://github.com/user-attachments/files/16578104/PR_1231_metric_profile_tests_openshift.zip)

File: tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+        file.write(data)
+
+    response = delete_metric_profile(input_json_file)
+    print("delete metric profile = ", response.status_code)
+    response = create_metric_profile(json_file)
+
+    data = response.json()
+    print(data['message'])
+
+    assert response.status_code == expected_status_code, \
```
[the remark this answers is not in the document]
Comment (shreyabiradar07) [reply tests/scripts/local_monitoring_tests/rest_apis/test_create_metric_profile.py:269]: Included assertion check for error messages

**Review by chandrams — APPROVED** [review]
LGTM

**Pull request merged**